src/docfx/lib/log/SourceInfo{T}.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+namespace Microsoft.Docs.Build
+{
+    [DebuggerDisplay("{Value}")]
+    public sealed class SourceInfo<T> : SourceInfo
+    {
+        public readonly T Value;
+
+        internal override object GetValue() => Value;
+
+        public SourceInfo(T value, SourceInfo source = null)
+            : base(source?.File, source?.Line ?? 0, source?.Column ?? 0, source?.EndLine ?? 0, source?.EndColumn ?? 0)
+        {
+            Value = value;
+        }
+
+        public SourceInfo<T> WithValue(T value)
+        {
+            if (value == default)
+                return null;
+
+            return new SourceInfo<T>(value, this);
+        }
+
+        public override string ToString()
```

Comment:
who is relying on this `ToString()` overload?

Reply:
https://github.com/dotnet/docfx/blob/575c0ad2b242ac79947270f68cedd3e057ae33b0/src/docfx/Errors.cs#L198

Reply:
ok, then `DebuggerDisplay` is not needed given the `ToString()` overload. And the `ToString` implementation needs to check `null`

Reply:
ok